src/lib.rs

```diff
@@ -54,10 +54,10 @@ pub struct Renderer {
 
 impl Renderer {
     /// Creates a new renderer for the specified device.
-    pub fn new(device: &Device) -> Result<Self> {
+    pub fn new(device: &Device, surface_format: TextureFormat) -> Result<Self> {
```

Comment:
Consider introducing a `RendererOptions` struct and adding `surface_format` as a field there. That would allow us to extend the parameters/options in the future without breaking the interface.